spring-integration-core/src/main/java/org/springframework/integration/channel/MessageChannelReactiveUtils.java

```diff
@@ -60,37 +61,18 @@ else if (messageChannel instanceof PollableChannel) {
 	}
 
 	private static <T> Publisher<Message<T>> adaptSubscribableChannelToPublisher(SubscribableChannel inputChannel) {
-		return new SubscribableChannelPublisherAdapter<>(inputChannel);
+		EmitterProcessor<Message<T>> publisher = EmitterProcessor.create(1);
```

Comment:
hint: for performance reasons, you may want to allocate a bigger buffer (e.g. 32, or 256, or even use the default one from `EmitterProcessor.create()`.

Reply:
Yeah... We had such a discussion with @olegz and our position is to not lose messages in case of crash.
When messages sit upstream blocked, there is more guarantee that they survive restart.
When we buffer them on the emitter level, it looks there is a possibility that we lose them.
No? What am I missing?

Reply:
prefetch will keep 1 message in memory anyways (mmm... I think)

Let me test it...

Reply:
Ok, so, the thing is:
A processor is okay, but the downstream most probably will prefetch (e.g. `flatMap` or `concatMap`), so it will request more than the user is ready to process.

Even if they set prefetch to 1 (the lowest possible value), it still means that there will be 1 element dangling in memory and it will be lost on crash

An acknowledgment mechanism may help here

Reply:
That is true and we also discussed it with Artem and we'll need to deal with it at some point, but in the science of sensible defaults it is my strong belief there are only two `lowest` and `highest` ;), so we chose lowest

Reply:
I think we will take that risk 😄 
At least we have only one message dangling around.
We have many ways to ack messages.
Different story though...

Thank so far.

I have failing test in the `ReactiveStreamsConsumerTests`. It looks like I can't subscribe a second time, even if I use `Flux.defer()`.

Dare to Zoom? 😉